### DISCUSSION TITLE
ChartJs multiple datasets

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1216] [Q&A (Help)] [answered]
Hello all,

Can someone advise how can I get ChartJS to look like the one in the demo page:
![image](https://github.com/user-attachments/assets/7b30b67e-80b1-409a-a581-b3d01b90ec96)

Is there a repository with the demo that I can look at that specific chart controller please, I am not able to get multiple datasets showing at the same time.

Thanks all

[a comment by an automated account is not shown]

**Reply by Read2uk** [reply]
Still not working, this is my setup method, I am not using the data() method

    public function setup()
    {
        $this->chart = new Chart();

        $months =[];
        $number_per_months =[];
        $total_fee_usd =[];
        foreach ($this->assignmentsPerMonth() as $numbers){
            $months[] = $numbers->month;
            $number_per_months[] = $numbers->number;
            $total_fee_usd[] = $numbers->total_fee_usd;
        }

        //Datasets
        $this->chart->dataset('Total Fee', 'line', array_reverse($total_fee_usd));
        $this->chart->dataset('Numbers', 'bar', array_reverse($number_per_months))
            ->color('rgba(205, 32, 31, 1)')
            ->backgroundColor('rgba(205, 32, 31, 0.4)');


        // MANDATORY. Set the labels for the dataset points
        $this->chart->labels(array_reverse($months));
        
        // RECOMMENDED. Set URL that the ChartJS library should call, to get its data using AJAX.
        $this->chart->load(backpack_url('charts/monthly-assignments'));

        // OPTIONAL
         $this->chart->minimalist(false);
         $this->chart->displayLegend(true);
    }

**Reply by tabacitu** [reply]
Heads-up @Read2uk - the AI doesn't answer to follow-ups right now. Only to your first question.

**Comment by tabacitu** [comment, marked as answer]
Exactly, the easiest way would be to take a look at how that's done in our demo. See:
- [the dashboard blade file here](https://github.com/Laravel-Backpack/demo/blob/main/resources/views/vendor/backpack/theme-tabler/dashboard.blade.php#L169-L178), which references:
- this [NewEntriesChartController](https://github.com/Laravel-Backpack/demo/blob/main/app/Http/Controllers/Admin/Charts/NewEntriesChartController.php);

**Reply by Read2uk** [reply]
Salut @tabacitu

The AI did a pretty good job! I just managed to sort the issues I was having! thank you
